doc/proposals/named-networks.md

```diff
@@ -0,0 +1,231 @@
++    // The ipRange is passed in so that the CR can be created if needed
++    pool, err := i.getPool(ctx, networkName, namespace, ipRange)
+     if err != nil {
+         return nil, err
+     }
+
+     firstIP, _, err := pool.ParseCIDR()
+     if err != nil {
+         return nil, err
+     }
+
+     return &KubernetesIPPool{i.client, i.containerID, firstIP, pool}, nil
+ }
+
+-func (i *KubernetesIPAM) getPool(ctx context.Context, name string, iprange string) (*whereaboutsv1alpha1.IPPool, error) {
++func (i *KubernetesIPAM) getPool(ctx context.Context, name string, namespace string, iprange string) (*whereaboutsv1alpha1.IPPool, error) {
+     pool := &whereaboutsv1alpha1.IPPool{
+         ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: i.namespace},
+     }
+-    if err := i.client.Get(ctx, types.NamespacedName{Name: name, Namespace: i.namespace}, pool); errors.IsNotFound(err) {
++    if err := i.client.Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, pool); errors.IsNotFound(err) {
+         // pool does not exist, create it
+         pool.ObjectMeta.Name = name
+         pool.Spec.Range = iprange
+         pool.Spec.Allocations = make(map[string]whereaboutsv1alpha1.IPAllocation)
+         if err := i.client.Create(ctx, pool); errors.IsAlreadyExists(err) {
+             // the pool was just created -- allow retry
+             return nil, &temporaryError{err}
+         } else if err != nil {
+             return nil, fmt.Errorf("k8s create error: %s", err)
+         }
+         // if the pool was created for the first time, trigger another retry of the allocation loop
+         // so all of the metadata / resourceVersions are populated as necessary by the `client.Get` call
+         return nil, &temporaryError{fmt.Errorf("k8s pool initialized")}
+     } else if err != nil {
+         return nil, fmt.Errorf("k8s get error: %s", err)
+     }
+     return pool, nil
+ }
+```
+
+### Summary
+
+whereabouts supports disabling the check for overlapping IP assignments, however it does not alllow actually configuring two identical ranges.
+
+This proposal (and the prototypical implementation in https://github.com/k8snetworkplumbingwg/whereabouts/pull/256) allows doing exactly that by introducing a new IPAM parameter `network_name`.
+
+### Discussions and Decisions
+
+See
+- https://github.com/k8snetworkplumbingwg/whereabouts/pull/256
+- https://github.com/k8snetworkplumbingwg/whereabouts/issues/50#issuecomment-874040513
+
+#### Migration toolkit
+
+Should the decision be made that scheme 1 or scheme 2 are implemented, we are facing a backwards-compatibility issue: whereabouts would not find existing `IPPool`s as they are stored under the CIDR-derived name.
+We propose to supply a small program (ideally also as OCI image for use in kubernetes `Job`s) that migrates an existing cluster from "legacy whereabouts" to the new, named and namespaced scheme.
+
+This tool would do roughly the following:
```

Comment:
Thanks Phillip! This looks great overall, and let's also have it attain the leader election lock so that it locks out whereabouts requests during this migration if possible.